polygeist/tools/cgeist/Lib/utils.h

```diff
@@ -48,6 +48,17 @@ enum class FunctionContext;
 
 namespace mlirclang {
 
+/// Whether the parent namespace is the sycl namespace, a namespace within it or
+/// not.
+enum class IsNamespaceSYCLResult {
+  /// The parent namespace is not the sycl namespace.
+  False = 0,
+  /// The parent namespace is the sycl namespace.
+  True,
+  /// The parent namespace is within the sycl namespace.
+  WithinSYCL,
+};
```

Comment:
```suggestion
enum class NamespaceKind {
  Other = 0,  /// namespace is not 'sycl'
  SYCL,       /// the 'sycl' namespace
  WithinSYCL, /// nested in the sycl namespace
};
```

Reply:
Prefer to use `NamespaceKind` for the typed enum name. If we need/want to track nested namespaces within the `sycl` namespace we can then just add enumerators, for example  `SYCLDetail` and `SYCLExtension`.